Support minizip-ng build

ljavorsk opened the pull request:
<!--
Thank you for your pull request.

If this fixes a GitHub issue, make sure to have a line saying 'Fixes #XXXX'
(without quotes) in the commit message.

Please update Signed-off-by info and read the common contributing guidelines
before you continue https://github.com/open-eid/.github/blob/master/CONTRIBUTING.md!
-->

Fedora has switched to `minizip-ng` (with the compat mode) and these patches fixed the failed build of `libdigidocpp`.

You can see more here: https://src.fedoraproject.org/rpms/libdigidocpp/pull-request/4

Signed-off-by: Lukas Javorsky <ljavorsk@redhat.com>

ljavorsk commented:
@metsma Could you please review? It is related to the issue https://github.com/open-eid/libdigidocpp/issues/569

metsma commented on CMakeLists.txt:
```diff
@@ -48,7 +48,7 @@ find_package(ZLIB REQUIRED)
 if(UNIX AND NOT APPLE)
     find_package(PkgConfig)
-    pkg_check_modules(MINIZIP minizip IMPORTED_TARGET minizip<2.0.0)
+    pkg_check_modules(MINIZIP minizip IMPORTED_TARGET minizip<5.0.0)
```
Why 5? do we need version restriction?

ljavorsk replied:
It was there before, and I wasn't sure if it served some purpose, so I left it there

metsma replied:
I think it was there to avoid pickup minizip-ng. My question is why version 5 or can we get rid of this condition?

ljavorsk replied:
From what I could test, version 3 is compatible after these two changes. I can test version 4 as well, but I cannot guarantee the upcoming ones.

ljavorsk replied:
@metsma I've tested it on version 4 and it's building, so I would leave it this way. Can you merge this?

ljavorsk replied:
Here is my build: https://copr.fedorainfracloud.org/coprs/ljavorsk/minizip-ng/build/6880707/

metsma replied:
Why not remove then the <5 condition? If new release is with major version number 5 then we need change code again

ljavorsk commented:
Changed to not check the version at all

ljavorsk commented:
Hi, can this MR be merged?

kristelmerilain commented:
We have to test it before we can merge it. 
If there are no problems during testing, then we will merge it as soon as possible.

metsma commented:
Can you please rebase to latest master to fix github actions and provide fedora builds with minizip-ng (needs probably installing minizip-ng-devel package?)

ljavorsk commented:
Rebased.
Builds are here: https://copr.fedorainfracloud.org/coprs/ljavorsk/minizip-ng/build/6880707/

It requires both `minizip-ng-compat-devel` and `zlib-ng-compat-devel` (same as I've changed here: https://src.fedoraproject.org/rpms/libdigidocpp/pull-request/4#request_diff). I left the `zlib-devel` because it's provided/obsoleted by `zlib-ng-compat-devel`.

metsma commented:
I was reffering to this project github action builds

https://github.com/open-eid/libdigidocpp/blob/master/.github/workflows/build.yml#L87

ljavorsk commented:
I've just added a new commit fixing the deps in Fedora. I used the old names because the new names work only since Fedora 40. However, the old names are working in Fedora 40 so don't worry

metsma commented on .github/workflows/build.yml:
```diff
@@ -84,7 +84,7 @@ jobs:
       run: |
         dnf install -y --setopt=install_weak_deps=False \
-          git gcc-c++ cmake rpm-build xml-security-c-devel zlib-devel vim-common doxygen boost-test swig python3-devel java-1.8.0-openjdk-devel xsd
+          git gcc-c++ cmake rpm-build xml-security-c-devel zlib-devel vim-common doxygen boost-test swig python3-devel java-1.8.0-openjdk-devel xsd minizip-devel
```
Isn't that minizip-ng-devel?
https://packages.fedoraproject.org/pkgs/minizip-ng/minizip-ng-devel/

ljavorsk replied:
No because once we switch to Fedora 40, the minizip-ng-devel is a different package than the minizip-ng-compat-devel.
When the minizip-devel is used it will be provided by `minizip-ng-devel` for Fedoras until 40 (which is okay, because it was wrongly named until that point, it should have been named minizip-ng-compat-devel) and since 40 it will be provided by `minizip-ng-compat-devel`.
